ls1intum/Artemis
# [Exam] Block Instructors from registering as students to exam

### Checklist

### Motivation and Context
Developers and testers often run into unexpected behaviour which is a result of instructors signing up as students to the exam. This breaks exam & test run related features. Therefore we want to block instructors of the course and administrators from signing up to an exam. 

Fixes #2489

### Description
Throw forbidden error on the client if the user tries to register an instructor or administrator to the exam. 

### Steps for Testing
1. Log in to Artemis
2. Navigate to an exam
3. Try to register an instructor of the course or an administrator to the exam
4. A forbidden error should be thrown

### Test Coverage
ExamResource: 99%
